src/main/java/org/jenkinsci/plugins/casc/configurationsource/ConfigurationSource.java

```diff
@@ -0,0 +1,31 @@
+package org.jenkinsci.plugins.casc.configurationsource;
+
+import hudson.ExtensionPoint;
+import jenkins.model.Jenkins;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * SPI For configurationsource source. All our sources are .yaml|.yml files, it's the method to which we resolve the value and
+ * where those values are stored we want to be able to configure.
+ */
+public abstract class ConfigurationSource implements ExtensionPoint {
+
+    public abstract Map<String, InputStream> listSources() throws IOException;
```

Comment:
No way! Please don't make InputStream as public api in 2018. Its not safe at all! 
- Where should it be closed? 
- How to make it NIO-compatible? 
- What if I have ready-to-use objects? Why should I bother with InputStream?

Please please don't!

Also see #119